lib/plaid/webhook.rb

```diff
@@ -0,0 +1,136 @@
+      @removed_transactions = fields['removed_transactions']
+      # present only for Error Response Webhook
+      @resolve = fields['resolve']
+    end
+
+    # Public: Share the type of Webhook this is from its code
+    #
+    # Returns String webhook type
+    def type
+      Webhook::CODEX[@code] || 'ErrorResponse'
+    end
+
+    # Public: Detect if the webhook is Initial Transaction Webhook. Occurs
+    # once the initial transaction pull has finished.
+    #
+    # Returns true if it is.
+    def initial_transaction?
+      code == Webhook::InitialTransaction
+    end
+
+    # Public: Detect if the webhook is Historical Transaction Webhook. Occurs
+    # once the historical transaction pull has completed, shortly after the
+    # initial transaction pull.
+    #
+    # Returns true if it is.
+    def historical_transaction?
+      code == Webhook::HistoricalTransaction
+    end
+
+    # Public: Detect if the webhook is Normal Transaction Webhook. Occurs at
+    # set intervals throughout the day as data is updated from the financial
+    # institutions.
+    #
+    # Returns true if it is.
+    def normal_transaction?
+      code == Webhook::NormalTransaction
+    end
+
+    # Public: Detect if the webhook is Removed Transaction Webhook. Occurs when
+    # transactions have been removed from our system.
+    #
+    # Returns true if it is.
+    def removed_transaction?
+      code == Webhook::RemovedTransaction
+    end
+
+    # Public: Detect if the webhook is User's Webhook Updated. Occurs when an
+    # user's webhook is updated via a PATCH request without credentials.
+    #
+    # Returns true if it is.
+    def user_webhook_updated?
+      code == Webhook::UserWebhookUpdated
+    end
+
+    # Public: Detect if the webhook is Error Response Webhook. Triggered when
+    # an error has occurred. Includes the relevant Plaid error code with
+    # details on both the error type and steps for error resolution.
+    #
+    # Returns true if it is.
+    def error_response?
+      Webhook::CODEX[code].nil?
+    end
+
+    # Public: Get transaction IDs that were removed.
+    #
+    # Returns Array[String] or nil
+    def removed_transactions_ids
+      @removed_transactions
+    end
+
+    # Public: Get a Plaid::Error instance if this is an Error Response Webhook
+    #
+    # Returns Plaid::Error or nil
+    def error
+      if error_response?
+        Plaid::PlaidError.new @code, @message, @resolve
+      end
+    end
+
+    # Public: Get a String representation of the transaction.
+    #
+    # Returns a String.
+    def inspect
+      "#<Plaid::Webhook type=#{type.inspect} code=#{code.inspect}, access_token=#{access_token.inspect}, " \
+      "total_transactions=#{total_transactions.inspect}, message=#{message.inspect}>"
+    end
+
+    # Public: Get a String representation of the transaction.
+    #
+    # Returns a String.
+    alias to_s inspect
+
+    private
+
+    Webhook::InitialTransaction    = 0
```

Comment:
I'd use ALL_CAPS for constants here. There's also duplication which can be avoided by using metaprogramming:

```ruby
codex = {}
['initial transaction', 'historical transaction'].each_with_index do |event, idx|
  name = event.split.map(&:upcase).join('_')
  codex[idx + 1] = name  
  const_set name, idx + 1
end

CODEX = codex.freeze
```